portindex2json/portindex2json.tcl

```diff
@@ -12,25 +12,80 @@
 
 package require json::write
 
+proc format_strings {strings} {
+	if {[llength $strings] > 1} {
+		foreach element $strings {
+			lappend new_strings "\"$element\""
+		}
+	} else {
+		set new_strings [list "\"$string\""]
+	}
+	return $new_strings
+}
+
+proc get_maintainer_array {maintainer} {
+	set detail_list [split $maintainer @]
+	if {[string first : [lindex $detail_list 0]] != -1} {
+		set email_list [split [lindex $detail_list 0] :]
+		set name [string trimright [lindex $email_list 1]]
+		set email(name) "\"$name\""
+		set email(domain) "\"[lindex $email_list 0]\""
+		::json::write indented false
+		set email_subobject [::json::write object {*}[array get email]]
+		::json::write indented true
+	} else {
+		set name [string trimright [lindex $detail_list 0]]
+		set email(name) "\"$name\""
+		set email(domain) "\"macports.org\""
+		::json::write indented false
+		set email_subobject [::json::write object {*}[array get email]]
+		::json::write indented true
+	}
+
+	set maintainer_array(github) "\"[lindex $detail_list 1]\""
+	set maintainer_array(email) $email_subobject
+
+	return [array get maintainer_array]
+}
+
+
 set fd [open [lindex $argv 0] r]
 chan configure $fd -encoding utf-8
 while {[gets $fd line] >= 0} {
-    if {[llength $line] != 2} {
-        continue
-    }
-    set len [lindex $line 1]
-    set line [read $fd $len]
-    array unset portinfo
-    array set portinfo $line
-    array unset json_portinfo
-    foreach key [array names portinfo] {
-        if { $key == "categories"} {
-            set json_portinfo($key) [::json::write array {*}$portinfo($key)]
-        } else {
-            set json_portinfo($key) [::json::write string $portinfo($key)]
-        }
-    }
-    lappend objects [::json::write object {*}[array get json_portinfo]]
+	if {[llength $line] != 2} {
+		continue
+	}
+	set len [lindex $line 1]
+	set line [read $fd $len]
+	array unset portinfo
+	array set portinfo $line
+	array unset json_portinfo
+	foreach key [array names portinfo] {
+		if { $key == "categories"  || $key=="variants" || $key=="depends_lib" || $key=="depends_build" } {
+			set json_portinfo($key) [::json::write array {*}[format_strings $portinfo($key)]]
+		} elseif { $key == "maintainers"} {
```

Comment:
I would probably try to put the `elseif` into a function that parses and writes maintainers, so that the main function stays easy enough to comprehend.

Reply:
Yes, that would be a really good improvement, thanks.

Reply:
I have changed it to use a procedure. Thanks